plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -408,6 +405,263 @@ export class GithubEntityProvider implements EntityProvider, EventSubscriber {
     );
   }
 
+  private async onRepoChange(event: RepositoryEvent) {
+    if (this.config.organization !== event.repository.organization) {
+      this.logger.debug(
+        `skipping repository event from organization ${event.repository.organization}`,
+      );
+      return;
+    }
+
+    const action = event.action;
+    switch (action) {
+      case 'archived':
+        await this.onRepoArchived(event as RepositoryArchivedEvent);
+        return;
+
+      // A repository was created.
+      case 'created':
+        // skip these events
+        return;
+
+      case 'deleted':
+        await this.onRepoDeleted(event as RepositoryDeletedEvent);
+        return;
+
+      case 'edited':
+        await this.onRepoEdited(event as RepositoryEditedEvent);
+        return;
+
+      // The visibility of a repository was changed to `private`.
+      case 'privatized':
+        // skip these events
+        return;
+
+      // The visibility of a repository was changed to `public`.
+      case 'publicized':
+        // skip these events
+        return;
+
+      case 'renamed':
+        await this.onRepoRenamed(event as RepositoryRenamedEvent);
+        return;
+
+      case 'transferred':
+        await this.onRepoTransferred(event as RepositoryTransferredEvent);
+        return;
+
+      case 'unarchived':
+        await this.onRepoUnarchived(event as RepositoryUnarchivedEvent);
+        return;
+
+      default: // should never be reached
+        this.logger.warn(
+          `Missing implementation for event of topic repository with action ${action}`,
+        );
+    }
+  }
+
+  /**
+   * A repository was archived.
+   *
+   * Removes all entities associated with the repository.
+   *
+   * @param event - The repository archived event.
+   * @private
+   */
+  private async onRepoArchived(event: RepositoryArchivedEvent) {
+    const repository = this.createRepoFromEvent(event);
+    await this.removeEntitiesForRepo(repository);
```

Comment:
should this action be optional (and set from app-config) in case that's not the behaviour that everyone expects?

referencing this comment here: https://github.com/backstage/backstage/issues/21906#issuecomment-1976830336

Reply:
Not at the moment. While there was interest in such an option, this would require changes to other parts like the full refresh, config/interface, etc., and would be scope creep to this change. Instead, this should be a separate issue and PR that changes all places.

Reply:
The "full refresh" would remove them already. And that is not configurable so far. It seems reasonable to provide the choice, though.

Reply:
I am ok with either outcomes, just a bit unsure if others would consider this some sort of breaking change or would be surprised if their archived repos are gone from their catalog.

Reply:
This is [the current behavior of the full refresh](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts#L268).

If there is any issue with that not functioning properly/reliably/..., it is a bug.

Reply:
ok all good